README.md

```diff
@@ -12,33 +12,49 @@ features from the kit.
+- **Messages**, serialized as `.ndjson` files, are the reference: a given feature
+  from the kit, executed using its dedicated step definitions, **must** emit the
   corresponding Messages
 
 The Messages from the kit are generated using
 [fake-cucumber](https://github.com/cucumber/fake-cucumber), which is used here
 as a reference implementation of the Messages protocol.
 
-## Contributing
+## Adoption
 
-This repo is composed of a [devkit](./devkit/), and packages to deliver the kit
-using managers like NPM or rubygem.
+To adopt the compatibility kit when writing a Cucumber implementation, an implementer must:
 
-### The devkit
+### Prerequisites
 
-The [devkit](./devkit/) is a set of tools to generate new Messages in order to
-extend the kit, or to update it after the protocol has been changed.
+First, make sure your implementation is sufficiently testable - you should be able to execute a test run programatically for a given set of feature files and step definitions, and obtain the Messages output (via a formatter or some other means) to make assertions on.
 
-### The packages
+### Setup
 
-At the moment, there are two packages: a NPM one, and a ruby gem.
+Pull in the latest version of the CCK package for your platform. If there isn't one yet, consider just copying over the files you need from the [devkit](./devkit/) for now - you can figure out the publishing later once it's all working.
 
-- the [NPM package](./javascript/) bundles the kit with step definitions
-  implemented in TypeScript.
-- the [ruby gem](./ruby), in addition of the kit and its step definitions
-  written in ruby, also provides a few helpers to make it easier to work with
-  the CCK.
+Next, for each feature in the CCK suite, write step definitions that work with your implementation - these should be _equivalent_ to the reference step definitions from the devkit, but with the appropriate language, syntax etc.
+
+### Testing
+
+Now you can build your CCK test. For each feature in the CCK suite, it should:
+
+1. Execute a test run, scoped to:
+  - Only the feature file in question
+  - Only the step definition file that you wrote for this feature
+2. Capture the Messages output of that test run
+3. Load the reference messages from the `.ndjson` file and assert that the output you captured matches it
+
+### Notes and exceptions
+
+When we say "matches" above, that's heavily qualified - there are many individual fields that will vary from run to run, like generated identifiers, timestamps, durations and file URLs. You can freely exclude these fields from your assertion - the key thing to capture is the number and order of the messages, and the content that would be consistent across runs. It's also not unheard of for implementations to fix up the order of messages so it matches the CCK, although it's best to try to match it naturally if you can.
```

Comment:
type, number and order of the messages - again as we've found out :D type is sometimes not correct haha

Reply:
Done